src/plonk_prove.js

```diff
@@ -48,7 +48,7 @@ export default async function plonk16Prove(zkeyFileName, witnessFileName, logger
         throw new Error(`Invalid witness length. Circuit: ${zkey.nVars}, witness: ${wtns.nWitness}, ${zkey.nAdditions}`);
     }
 
-    const curve = await getCurve(zkey.q);
+    const curve = zkey.curve;
```

Comment:
```suggestion
    const curve = zkey.curve || await getCurve(zkey.q);
```

@nalinbhardwaj I'm wondering if we should put this safeguard in place. Is there **any** conceivable way that the zkey could be missing the `curve` property?

Reply:
I don't think so - the zkey object gets set in `zkeyUtils.readHeader` and there isn't an early return/another source for the zkey object.

Reply:
Regardless, `zkey.q` is also set in the same place as `curve` so this line would be still be undefined if `zkey.curve` were to be somehow undefined

Reply:
👍  Sounds good.